### PR TITLE
[Translation] Fix extraction when dealing with VariadicPlaceholder parameters

### LINKED ISSUE #48422: [Translation] [6.2] debug:translation and translation:extract broken
### Symfony version(s) affected

6.2

### Description

The translation related console commands fail when updating to 6.2.

`bin/console debug:translation en`
`bin/console translation:extract --force en`

```console
In CallLike.php line 36:

  [AssertionError (1)]
  assert(!$this->isFirstClassCallable())
```

### How to reproduce

Install Symfony skeleton
```
symfony new translation-reproducer --version="6.2.*" --webapp
```

Run one of the commands
```
bin/console debug:translation en -vvv
bin/console translation:extract --force en -vvv
```

### Possible Solution

I believe the fix could start at L66 in AbstractVisitor.php:

```diff
-$args = $node instanceof Node\Expr\CallLike ? $node->getArgs() : $node->args;
+$args = $node instanceof Node\Expr\CallLike && $node->isFirstClassCallable() === false ? $node->getArgs() : $node->args;
```

But this leads to the next error:
```
In AbstractVisitor.php line 70:

  [ErrorException]
  Warning: Undefined property: PhpParser\Node\VariadicPlaceholder::$name
```

Which again could be fixed by
```diff
-if (null !== $arg->name) {
+if (property_exists($arg, 'name') === false || null !== $arg->name) {
```

But this leads to a few new similar errors and soon a deadend at least for me, because I don't really understand what the code is supposed to do.

### Additional Context

System info:

```
transrep git:(main) bin/console about
 -------------------- ---------------------------------
  Symfony
 -------------------- ---------------------------------
  Version              6.2.0
  Long-Term Support    No
  End of maintenance   07/2023 (in +242 days)
  End of life          07/2023 (in +242 days)
 -------------------- ---------------------------------
  Kernel
 -------------------- ---------------------------------
  Type                 App\Kernel
  Environment          dev
  Debug                true
  Charset              UTF-8
  Cache directory      ./var/cache/dev (7.7 MiB)
  Build directory      ./var/cache/dev (7.7 MiB)
  Log directory        ./var/log (26 KiB)
 -------------------- ---------------------------------
  PHP
 -------------------- ---------------------------------
  Version              8.1.13
  Architecture         64 bits
  Intl locale          en_US_POSIX
  Timezone             UTC (2022-12-01T11:42:30+00:00)
  OPcache              true
  APCu                 false
  Xdebug               false
 -------------------- ---------------------------------
```

Stack traces:

```
transrep git:(main) bin/console translation:extract --force en -vvv

Translation Messages Extractor and Dumper
=========================================

 // Generating "en" translation files for "default directory"

 // Parsing templates...


In CallLike.php line 36:

  [AssertionError (1)]
  assert(!$this->isFirstClassCallable())


Exception trace:
  at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/Node/Expr/CallLike.php:36
 assert() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/Node/Expr/CallLike.php:36
 PhpParser\Node\Expr\CallLike->getArgs() at /Users/gndk/dev/transrep/vendor/symfony/translation/Extractor/Visitor/AbstractVisitor.php:66
 Symfony\Component\Translation\Extractor\Visitor\AbstractVisitor->hasNodeNamedArguments() at /Users/gndk/dev/transrep/vendor/symfony/translation/Extractor/Visitor/TransMethodVisitor.php:40
 Symfony\Component\Translation\Extractor\Visitor\TransMethodVisitor->enterNode() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:123
 PhpParser\NodeTraverser->traverseNode() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:146
 PhpParser\NodeTraverser->traverseNode() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:146
 PhpParser\NodeTraverser->traverseNode() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:91
 PhpParser\NodeTraverser->traverse() at /Users/gndk/dev/transrep/vendor/symfony/translation/Extractor/PhpAstExtractor.php:56
 Symfony\Component\Translation\Extractor\PhpAstExtractor->extract() at /Users/gndk/dev/transrep/vendor/symfony/translation/Extractor/ChainExtractor.php:48
 Symfony\Component\Translation\Extractor\ChainExtractor->extract() at /Users/gndk/dev/transrep/vendor/symfony/framework-bundle/Command/TranslationUpdateCommand.php:395
 Symfony\Bundle\FrameworkBundle\Command\TranslationUpdateCommand->extractMessages() at /Users/gndk/dev/transrep/vendor/symfony/framework-bundle/Command/TranslationUpdateCommand.php:200
 Symfony\Bundle\FrameworkBundle\Command\TranslationUpdateCommand->execute() at /Users/gndk/dev/transrep/vendor/symfony/console/Command/Command.php:312
 Symfony\Component\Console\Command\Command->run() at /Users/gndk/dev/transrep/vendor/symfony/console/Application.php:1038
 Symfony\Component\Console\Application->doRunCommand() at /Users/gndk/dev/transrep/vendor/symfony/framework-bundle/Console/Application.php:88
 Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() at /Users/gndk/dev/transrep/vendor/symfony/console/Application.php:312
 Symfony\Component\Console\Application->doRun() at /Users/gndk/dev/transrep/vendor/symfony/framework-bundle/Console/Application.php:77
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /Users/gndk/dev/transrep/vendor/symfony/console/Application.php:168
 Symfony\Component\Console\Application->run() at /Users/gndk/dev/transrep/vendor/symfony/runtime/Runner/Symfony/ConsoleApplicationRunner.php:54
 Symfony\Component\Runtime\Runner\Symfony\ConsoleApplicationRunner->run() at /Users/gndk/dev/transrep/vendor/autoload_runtime.php:29
 require_once() at /Users/gndk/dev/transrep/bin/console:11

translation:extract [--prefix [PREFIX]] [--format [FORMAT]] [--dump-messages] [--force] [--clean] [--domain [DOMAIN]] [--sort [SORT]] [--as-tree [AS-TREE]] [--] <locale> [<bundle>]
```

```
transrep git:(main) bin/console debug:translation en -vvv

In CallLike.php line 36:

  [AssertionError (1)]
  assert(!$this->isFirstClassCallable())


Exception trace:
  at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/Node/Expr/CallLike.php:36
 assert() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/Node/Expr/CallLike.php:36
 PhpParser\Node\Expr\CallLike->getArgs() at /Users/gndk/dev/transrep/vendor/symfony/translation/Extractor/Visitor/AbstractVisitor.php:66
 Symfony\Component\Translation\Extractor\Visitor\AbstractVisitor->hasNodeNamedArguments() at /Users/gndk/dev/transrep/vendor/symfony/translation/Extractor/Visitor/TransMethodVisitor.php:40
 Symfony\Component\Translation\Extractor\Visitor\TransMethodVisitor->enterNode() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:123
 PhpParser\NodeTraverser->traverseNode() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:146
 PhpParser\NodeTraverser->traverseNode() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:146
 PhpParser\NodeTraverser->traverseNode() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /Users/gndk/dev/transrep/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:91
 PhpParser\NodeTraverser->traverse() at /Users/gndk/dev/transrep/vendor/symfony/translation/Extractor/PhpAstExtractor.php:56
 Symfony\Component\Translation\Extractor\PhpAstExtractor->extract() at /Users/gndk/dev/transrep/vendor/symfony/translation/Extractor/ChainExtractor.php:48
 Symfony\Component\Translation\Extractor\ChainExtractor->extract() at /Users/gndk/dev/transrep/vendor/symfony/framework-bundle/Command/TranslationDebugCommand.php:346
 Symfony\Bundle\FrameworkBundle\Command\TranslationDebugCommand->extractMessages() at /Users/gndk/dev/transrep/vendor/symfony/framework-bundle/Command/TranslationDebugCommand.php:170
 Symfony\Bundle\FrameworkBundle\Command\TranslationDebugCommand->execute() at /Users/gndk/dev/transrep/vendor/symfony/console/Command/Command.php:312
 Symfony\Component\Console\Command\Command->run() at /Users/gndk/dev/transrep/vendor/symfony/console/Application.php:1038
 Symfony\Component\Console\Application->doRunCommand() at /Users/gndk/dev/transrep/vendor/symfony/framework-bundle/Console/Application.php:88
 Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() at /Users/gndk/dev/transrep/vendor/symfony/console/Application.php:312
 Symfony\Component\Console\Application->doRun() at /Users/gndk/dev/transrep/vendor/symfony/framework-bundle/Console/Application.php:77
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /Users/gndk/dev/transrep/vendor/symfony/console/Application.php:168
 Symfony\Component\Console\Application->run() at /Users/gndk/dev/transrep/vendor/symfony/runtime/Runner/Symfony/ConsoleApplicationRunner.php:54
 Symfony\Component\Runtime\Runner\Symfony\ConsoleApplicationRunner->run() at /Users/gndk/dev/transrep/vendor/autoload_runtime.php:29
 require_once() at /Users/gndk/dev/transrep/bin/console:11

debug:translation [--domain [DOMAIN]] [--only-missing] [--only-unused] [--all] [--] <locale> [<bundle>]
```

### PR DESCRIPTION
**Pull request by Kocal**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix #48422 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

I don't have the time for now to check if the command has tests or not, but dealing with raw args with + filtering for `Node\Arg` instance make the `extract` command working again.

The issue is "caused by" https://github.com/symfony/symfony/blob/6.3/src/Symfony/Bridge/Twig/Extension/TranslationExtension.php#L71, the extract command is not able to deal with VariadicPlaceholder node. Changing to `[$this, 'trans']` works, but this is not the wanted solution. 

**Before:**

```
symfony console translation:extract fr --force -vvv

Translation Messages Extractor and Dumper
=========================================

 // Generating "fr" translation files for "default directory"

 // Parsing templates...


In CallLike.php line 36:

  [AssertionError (1)]
  assert(!$this->isFirstClassCallable())


Exception trace:
  at /Users/kocal/workspace/chronos/vendor/nikic/php-parser/lib/PhpParser/Node/Expr/CallLike.php:36
 assert() at /Users/kocal/workspace/chronos/vendor/nikic/php-parser/lib/PhpParser/Node/Expr/CallLike.php:36
 PhpParser\Node\Expr\CallLike->getArgs() at /Users/kocal/workspace/chronos/vendor/symfony/translation/Extractor/Visitor/AbstractVisitor.php:66
 Symfony\Component\Translation\Extractor\Visitor\AbstractVisitor->hasNodeNamedArguments() at /Users/kocal/workspace/chronos/vendor/symfony/translation/Extractor/Visitor/TransMethodVisitor.php:40
 Symfony\Component\Translation\Extractor\Visitor\TransMethodVisitor->enterNode() at /Users/kocal/workspace/chronos/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:123
 PhpParser\NodeTraverser->traverseNode() at /Users/kocal/workspace/chronos/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /Users/kocal/workspace/chronos/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at /Users/kocal/workspace/chronos/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:146
 PhpParser\NodeTraverser->traverseNode() at /Users/kocal/workspace/chronos/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /Users/kocal/workspace/chronos/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at /Users/kocal/workspace/chronos/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:146
 PhpParser\NodeTraverser->traverseNode() at /Users/kocal/workspace/chronos/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /Users/kocal/workspace/chronos/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at /Users/kocal/workspace/chronos/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /Users/kocal/workspace/chronos/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at /Users/kocal/workspace/chronos/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /Users/kocal/workspace/chronos/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at /Users/kocal/workspace/chronos/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at /Users/kocal/workspace/chronos/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php:91
 PhpParser\NodeTraverser->traverse() at /Users/kocal/workspace/chronos/vendor/symfony/translation/Extractor/PhpAstExtractor.php:56
 Symfony\Component\Translation\Extractor\PhpAstExtractor->extract() at /Users/kocal/workspace/chronos/vendor/symfony/translation/Extractor/ChainExtractor.php:48
 Symfony\Component\Translation\Extractor\ChainExtractor->extract() at /Users/kocal/workspace/chronos/vendor/symfony/framework-bundle/Command/TranslationUpdateCommand.php:395
 Symfony\Bundle\FrameworkBundle\Command\TranslationUpdateCommand->extractMessages() at /Users/kocal/workspace/chronos/vendor/symfony/framework-bundle/Command/TranslationUpdateCommand.php:200
 Symfony\Bundle\FrameworkBundle\Command\TranslationUpdateCommand->execute() at /Users/kocal/workspace/chronos/vendor/symfony/console/Command/Command.php:312
 Symfony\Component\Console\Command\Command->run() at /Users/kocal/workspace/chronos/vendor/symfony/console/Application.php:1038
 Symfony\Component\Console\Application->doRunCommand() at /Users/kocal/workspace/chronos/vendor/symfony/framework-bundle/Console/Application.php:88
 Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() at /Users/kocal/workspace/chronos/vendor/symfony/console/Application.php:312
 Symfony\Component\Console\Application->doRun() at /Users/kocal/workspace/chronos/vendor/symfony/framework-bundle/Console/Application.php:77
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /Users/kocal/workspace/chronos/vendor/symfony/console/Application.php:168
 Symfony\Component\Console\Application->run() at /Users/kocal/workspace/chronos/vendor/symfony/runtime/Runner/Symfony/ConsoleApplicationRunner.php:54
 Symfony\Component\Runtime\Runner\Symfony\ConsoleApplicationRunner->run() at /Users/kocal/workspace/chronos/vendor/autoload_runtime.php:29
```

**After:**
```
symfony console translation:extract fr --force -vvv

Translation Messages Extractor and Dumper
=========================================

 // Generating "fr" translation files for "default directory"

 // Parsing templates...

 // Loading translation files...

 // Writing files...


 [OK] Translation files were successfully updated.

```

### CODE REVIEW

**Comment by gndk** [comment]
> I don't have the time for now to check if the command has tests or not

Tests for the commands are here in the `FramewordBundle`:

[TranslationDebugCommandTest](https://github.com/symfony/symfony/blob/6.2/src/Symfony/Bundle/FrameworkBundle/Tests/Command/TranslationDebugCommandTest.php)
[TranslationUpdateCommandTest](https://github.com/symfony/symfony/blob/6.2/src/Symfony/Bundle/FrameworkBundle/Tests/Command/TranslationUpdateCommandTest.php)

Curious why they did not catch this 🤔 

> The issue is "caused by" https://github.com/symfony/symfony/blob/6.3/src/Symfony/Bridge/Twig/Extension/TranslationExtension.php#L71

Maybe because they are not run with actual twig templates as fixtures?

File: src/Symfony/Component/Translation/Extractor/Visitor/AbstractVisitor.php [hunk 1]
```diff
@@ -52,22 +52,21 @@ protected function getStringArguments(Node\Expr\CallLike|Node\Attribute|Node\Exp
             return [];
         }
 
-        /** @var Node\Arg $arg */
-        $arg = $args[$index];
-        if (!$result = $this->getStringValue($arg->value)) {
-            return [];
+        if (($arg = $args[$index]) instanceof Node\Arg) {
```
Comment (kniziol) [inline src/Symfony/Component/Translation/Extractor/Visitor/AbstractVisitor.php:55]:
```suggestion
        $args = $node instanceof Node\Expr\CallLike ? $node->getRawArgs() : $node->args;

        if (($arg = $args[$index]) instanceof Node\Arg) {
```
The `$args` variable has been moved from 45 line

File: src/Symfony/Component/Translation/Extractor/Visitor/AbstractVisitor.php [hunk 2]
```diff
@@ -42,7 +42,7 @@ protected function addMessageToCatalogue(string $message, ?string $domain, int $
 
     protected function getStringArguments(Node\Expr\CallLike|Node\Attribute|Node\Expr\New_ $node, int|string $index, bool $indexIsRegex = false): array
     {
-        $args = $node instanceof Node\Expr\CallLike ? $node->getArgs() : $node->args;
+        $args = $node instanceof Node\Expr\CallLike ? $node->getRawArgs() : $node->args;
```
Comment (kniziol) [inline src/Symfony/Component/Translation/Extractor/Visitor/AbstractVisitor.php]:
```suggestion
```
Looks like `$args` variable is unnecessary before the `if (\is_string($index)) {` condition

**Comment by nicolas-grekas** [comment]
Thank you @Kocal.

**Pull request merged**